.github/workflows/main.yml

```diff
@@ -0,0 +1,23 @@
+name: CI
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: ["*"]
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: true
+      - name: Update local toolchain
+        run: |
+          rustup update
+          rustup component add clippy
+          rustup toolchain install nightly --allow-downgrade -c rustfmt
```

Comment:
If you don't mind using nightly for just rustfmt, you get access to a lot of nice options.